src/transaction.ts

```diff
@@ -134,7 +135,9 @@ export class StacksTransaction {
     const bufferArray: BufferArray = new BufferArray();
 
     bufferArray.appendByte(this.version);
-    bufferArray.appendHexString(this.chainId);
+    const chainIdBuffer = Buffer.alloc(4);
```

Comment:
We could probably write a helper function for appending uint32's. But this works for now.